citesphere/src/main/webapp/WEB-INF/views/auth/authorities/create.html

```diff
@@ -31,13 +36,20 @@
     });
 
     $("#searchAuthor").click(function() {
+    	$("#conceptpowerAuthoritySearchResult").empty();
         $("#selectAuthorityModel").on('hidden.bs.modal', function(e) {
             $('#selectAuthorityModel a:first').tab('show');
         })
-        $("#searchAuthorSpinner").show();
+        
+        $('#conceptpowerAuthoritiesTabContent').toggleClass('greyout');
+        $("#authoritySpinner").show();
         $('#conceptpowerAuthority-pagination-top').twbsPagination('destroy');
-        getConceptpowerAuthorities('Author', 'Author', 0)
-        $("#searchAuthorSpinner").hide();
+        
+        setTimeout(function() {getConceptpowerAuthorities('Author', 'Author', 0);}, 2000);
+        setTimeout(function() {
+			  $('#authoritySpinner').hide()
+			}, 3000);
```

Comment:
what if requesting the authorities takes longer than 1 sec?

Reply:
I had tested for a few heavy searches and given an average time of 2 seconds, but can change it if it is something more or may be see if there's an alternative for this time to change dynamically

Reply:
Instead you should be polling. set the timeout for 1 second or so and if there are results yet, recursively call the same function, until you get a result. Just because in your tests it didn't take longer than 2 seconds, doesn't mean it won't ever.